File: app/src/main/java/com/seafile/seadroid2/framework/model/sdoc/SDocTagModel.java
package com.seafile.seadroid2.framework.model.sdoc;

import android.os.Parcel;
import android.os.Parcelable;

public class SDocTagModel implements Parcelable {
    public String name;
    public String id;
    public String color;

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.name);
        dest.writeString(this.id);
        dest.writeString(this.color);
    }

    public SDocTagModel() {
    }

    protected SDocTagModel(Parcel in) {
        this.name = in.readString();
        this.id = in.readString();
        this.color = in.readString();
    }

    public static final Parcelable.Creator<SDocTagModel> CREATOR = new Parcelable.Creator<SDocTagModel>() {
        @Override
        public SDocTagModel createFromParcel(Parcel source) {
            return new SDocTagModel(source);
        }

        @Override
        public SDocTagModel[] newArray(int size) {
            return new SDocTagModel[size];
        }
    };
}

File: app/src/main/java/com/seafile/seadroid2/framework/model/sdoc/SDocTagModel.kt
package com.seafile.seadroid2.framework.model.sdoc

import android.os.Parcel
import android.os.Parcelable

class SDocTagModel() : Parcelable {
    @JvmField
    var name: String? = null

    @JvmField
    var id: String? = null

    @JvmField
    var color: String? = null

    override fun describeContents(): Int = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeString(id)
        dest.writeString(color)
    }

    private constructor(parcel: Parcel) : this() {
        name = parcel.readString()
        id = parcel.readString()
        color = parcel.readString()
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<SDocTagModel> = object : Parcelable.Creator<SDocTagModel> {
            override fun createFromParcel(source: Parcel): SDocTagModel = SDocTagModel(source)
            override fun newArray(size: Int): Array<SDocTagModel?> = arrayOfNulls(size)
        }
    }
}
